rfc/rfc023-x509credential.md

```diff
@@ -0,0 +1,453 @@
+
+* [X.509 Certificate Standard](https://datatracker.ietf.org/doc/html/rfc5280)
+* [JSON Web Signature (JWS)](https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.6)
+* [did:x509 method specification](https://trustoverip.github.io/tswg-did-x509-method-specification/), with modifications
+* W3C Verifiable Credentials Data Model
+
+### x509 certificates, a brief introduction
+
+The structure of an x509 certificate is defined by the X.509 standard. An x509 certificate contains information about
+the holder and is signed by a Certificate Authority (CA). The CA is a trusted party that is capable of verifying the
+identity of the holder of the certificate. The CA signs the certificate with its own private key. The holder of the
+certificate can then use the public key of the CA to verify the signature of the certificate. This way the holder of the
+certificate can prove that the certificate is valid and that the information in the certificate is correct.
+
+The verifier of a x509 certificate can then trust the information in the certificate by verifying the signature of the
+certificate chain of the certificate. The verifier can then trust the information in the certificate by trusting the CA
+that signed the certificate.
+
+The chain of certificates can be viewed as a hierarchy, where the root certificate is the certificate is trusted, and
+signing is delegated to intermediate certificates. The root certificate is the certificate that is trusted by the
+holder. The holder maintains a list of trusted CAs that the holder trusts. The holder can then
+verify the signature of the certificate chain by verifying the signature of the CA that signed the intermediate
+certificate and the intermediate certificates that lead to the signing certificate.
+
+```asciidoc
+┌────────────────────┐
+│        CA          │
+└─────────┬──────────┘
+          │           
+┌─────────▼──────────┐
+│    Intermediate    │
+└─────────┬──────────┘
+          │           
+┌─────────▼──────────┐
+│    Intermediate    │
+└─────────┬──────────┘
+          │           
+┌─────────▼──────────┐
+│Signing Certificate │
+└────────────────────┘
+```
+
+### Using x509 for signing JWEs
+
+The JWE is a standard that is used to sign and encrypt JSON objects. Thus standard allows for the signing and encryption
+of JSON objects with certificates part of the a certificate chain. This allows for the signing of JSON objects with the
+private key of the certificate and the verification of the signature with the public key of the certificate, and the
+verification of the certificate chain with the public key of the CA. This is done by using the following headers fields:
+
+* x5c, the certificate chain as a list of base64 encoded certificates in the DER format, with the signing certificate
+  first and the root certificate last.
+* x5t, the thumbprint of the signing certificate.
+* x5t#S256, the thumbprint of the signing certificate as a SHA256 hash.
```

Comment:
I think at least one is required, right?

Reply:
probably don't accept `sha-1` -> must use `x5t#S256`? 
linked to nuts-foundation/go-didx509-toolkit/issues/53